README.md

```diff
@@ -32,17 +32,38 @@ Arrow is a set of technologies that enable big-data systems to process and move
 Initial implementations include:
 
  - [The Arrow Format](https://github.com/apache/arrow/tree/master/format)
- - [Arrow Structures and APIs in C++](https://github.com/apache/arrow/tree/master/cpp)
- - [Arrow Structures and APIs in Java](https://github.com/apache/arrow/tree/master/java)
+ - [Java implementation](https://github.com/apache/arrow/tree/master/java)
+ - [C++ implementation](https://github.com/apache/arrow/tree/master/cpp)
+ - [Python interface to C++ libraries](https://github.com/apache/arrow/tree/master/python)
 
-Arrow is an [Apache Software Foundation](www.apache.org) project. More info can be found at [arrow.apache.org](http://arrow.apache.org).
+Arrow is an [Apache Software Foundation](www.apache.org) project. Learn more at
+[arrow.apache.org](http://arrow.apache.org).
+
+#### What's in the Arrow libraries?
+
+The reference Arrow implementations contain a number of distinct software
+components:
+
+- Columnar vector/array and table row batch containers supporting nested data
```

Comment:
Maybe we should drop `DataFrame` (or `superset of a DataFrame`) somewhere. That's at least what people in the PyData space are more used too.